# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/PDMFactory.cpp
# Commit: e0db5954e844
# Full Hash: e0db5954e844d59ad70d8636f5faa99232fd005a
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-10-22 14:54:09
# Description:
#   Bug 1672553 - Don't use AudioTrimmer if audio decoder creation failed. r=mattwoodrow
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D94400
# ==============================================================================

diff -r 46bb9fa2d70c -r e0db5954e844 dom/media/platforms/PDMFactory.cpp
--- a/dom/media/platforms/PDMFactory.cpp	Thu Oct 22 01:03:12 2020 +0000
+++ b/dom/media/platforms/PDMFactory.cpp	Thu Oct 22 01:16:05 2020 +0000
@@ -271,7 +271,7 @@
 
   if (config.IsAudio()) {
     m = aPDM->CreateAudioDecoder(aParams);
-    if (!aParams.mNoWrapper.mDontUseWrapper) {
+    if (m && !aParams.mNoWrapper.mDontUseWrapper) {
       m = new AudioTrimmer(m.forget(), aParams);
     }
     return m.forget();
